source/passthrough.d.ts

```diff
@@ -0,0 +1,15 @@
+/**
+A passthrough type that allows some types to be used to extend an interface, which originally can't.
```

Comment:
It would be useful to document exactly what types.

Reply:
I've added a [use-case](https://github.com/sindresorhus/type-fest/pull/188/files#diff-f7c49e21c2e22539ee8f81d4a42709bbd8700866e88aa1aa55a3d86cfaca322dR4-R5) (more if I can think out of later), hopefully it would easier to understand.